bgpd/bgp_route.c

```diff
@@ -11096,53 +11118,105 @@ static int bgp_table_stats(struct vty *vty, struct bgp *bgp, afi_t afi,
 #endif
 		case BGP_STATS_ASPATH_TOTHOPS:
 		case BGP_STATS_ASPATH_TOTSIZE:
-			vty_out(vty, "%-30s: ", table_stats_strs[i]);
-			vty_out(vty, "%12.2f",
-				ts.counts[i]
-					? (float)ts.counts[i]
-						  / (float)ts.counts
-							    [BGP_STATS_ASPATH_COUNT]
-					: 0);
+			if (!json) {
+				snprintf(temp_buf, sizeof(temp_buf), "%12.2f",
+					 ts.counts[i] ? (float)ts.counts[i] /
+					 (float)ts.counts[BGP_STATS_ASPATH_COUNT] : 0);
+				vty_out(vty, "%-30s: %s", table_stats_strs[i][0], temp_buf);
+			} else {
```

Comment:
When we have output that are numbers they should be *numbers* not strings.  Please fix this across all output

Reply:
this is an issue for me now.
I think I should use `json_object_new_double()` API from libjson.
so that implies an extension of lib/json.c . right?

Reply:
yes.

Reply:
done. however, as I am newbie on json, and even if I basically tested it, a second pair of eyes is welcome.